Objects/abstract.c

```diff
@@ -819,6 +819,20 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (op_slot == NB_SLOT(nb_rshift) && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
```

Comment:
Moving the review comment about checking for *exactly* the print builtin to the latest commit:

```
_Py_IDENTIFIER(print);
builtins = PyEval_GetBuiltins();
print_ref = _PyDict_GetItemId(builtins, &PyId_print);
```

and then check for `v == print_ref`. 

(Only don't lay it out exactly like that, adjust it to fit the code - the above is just to show the key APIs you need)